api/class-wc-calypso-bridge-send-invoice-controller.php

```diff
@@ -0,0 +1,127 @@
+					'type'        => 'integer',
+				)
+			),
+			array(
+				'methods'             => WP_REST_SERVER::EDITABLE,
+				'callback'            => array( $this, 'send_invoice' ),
+				'permission_callback' => array( $this, 'permissions_check' ),
+			)
+		) );
+	}
+
+	/**
+	 * Makes sure the current user has permissions.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function permissions_check( $request ) {
+		$order = wc_get_order( (int) $request['id'] );
+		if ( $order && ! wc_rest_check_post_permissions( $this->post_type, 'read', $order->get_id() ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot view this resource.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+    
+	/**
+	 * Prepare a single order note output for response.
+	 *
+	 * @param WP_Comment $note Order note object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $note, $request ) {
+		$data = array(
+			'id'               => (int) $note->comment_ID,
+			'author'           => __( 'WooCommerce', 'woocommerce' ) === $note->comment_author ? 'system' : $note->comment_author,
+			'date_created'     => wc_rest_prepare_date_response( $note->comment_date ),
+			'date_created_gmt' => wc_rest_prepare_date_response( $note->comment_date_gmt ),
+			'note'             => $note->comment_content,
+			'customer_note'    => (bool) get_comment_meta( $note->comment_ID, 'is_customer_note', true ),
+		);
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		// Wrap the data in a response object.
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Send order invoice email.
+	 *
+	 * @param WP_REST_Request
+	 * @return WP_REST_Response
+	 *
+	 */
+	public function send_invoice( $request ) {
+		$order = wc_get_order( (int) $request['id'] );
+		if ( ! $order || $this->post_type !== $order->get_type() ) {
+			return new WP_Error( 'woocommerce_rest_order_invalid_id', __( 'Invalid order ID.', 'woocommerce' ), array( 'status' => 404 ) );
+		}
+
+		do_action( 'woocommerce_before_resend_order_emails', $order, 'customer_invoice' );
+		// Send the customer invoice email.
+		WC()->payment_gateways();
+		WC()->shipping();
+		WC()->mailer()->customer_invoice( $order );
+		// Note the event.
+		$note_id = $order->add_order_note( __( 'Order details manually sent to customer.', 'woocommerce' ), false, true );
+		if ( ! $note_id ) {
+			return new WP_Error( 'woocommerce_api_cannot_create_order_note', __( 'Cannot create order note, please try again.', 'woocommerce' ), array( 'status' => 500 ) );
+		}
+		$note = get_comment( $note_id );
+
+		do_action( 'woocommerce_after_resend_order_email', $order, 'customer_invoice' );
+		$response = $this->prepare_item_for_response( $note, $request );
```

Comment:
I'm not sure the note is a good response, since it might happen regardless of whether the email goes through, but since `WC()->mailer()->customer_invoice( $order );` doesn't return anything I guess this is the best option.

Reply:
Yeah - i know in some wpcom endpoints we return something simple like `success: true` - but from my cursory tour of the WC api - i don't see any current endpoints that don't return objects/arrays